pkg/agent/baker.go

```diff
@@ -256,24 +258,31 @@ func normalizeResourceGroupNameForLabel(resourceGroupName string) string {
 	if strings.HasSuffix(truncated, "-") ||
 		strings.HasSuffix(truncated, "_") ||
 		strings.HasSuffix(truncated, ".") {
-		if len(truncated) > 62 {
+		if len(truncated) > maxLen-1 {
 			return truncated[0:len(truncated)-1] + "z"
-		} else {
-			return truncated + "z"
 		}
+		return truncated + "z"
 	}
 	return truncated
 }
 
 func validateAndSetLinuxNodeBootstrappingConfiguration(config *datamodel.NodeBootstrappingConfiguration) {
 	// If using kubelet config file, disable DynamicKubeletConfig feature gate and remove dynamic-config-dir
 	// we should only allow users to configure from API (20201101 and later)
+	dockerShimFlags := []string{"--cni-bin-dir",
+		"--cni-cache-dir",
+		"--cni-conf-dir",
+		"--docker-endpoint",
+		"--image-pull-progress-deadline",
+		"--network-plugin",
+		"--network-plugin-mtu"}
```

Comment:
```suggestion
	dockerShimFlags := []string{
	    "--cni-bin-dir",
		"--cni-cache-dir",
		"--cni-conf-dir",
		"--docker-endpoint",
		"--image-pull-progress-deadline",
		"--network-plugin",
		"--network-plugin-mtu",
    }
```
nit: personal preference